test/integration/roles/test_service/tasks/main.yml

```diff
@@ -11,9 +11,9 @@
     - "install_result.mode == '0755'"
 
 - include: 'sysv_setup.yml'
-  when: ansible_distribution in ['RedHat', 'CentOS', 'ScientificLinux']
+  when: ansible_distribution in ['Fedora', 'RedHat', 'CentOS', 'ScientificLinux'] and (ansible_distribution_version|version_compare('6', '>=') and ansible_distribution_version|version_compare('7', '<'))
 - include: 'systemd_setup.yml'
-  when: ansible_distribution == 'Fedora'
+  when: ansible_distribution in ['Fedora', 'RedHat', 'CentOS', 'ScientificLinux'] and (ansible_distribution_version|version_compare('7', '>=') and ansible_distribution_version|version_compare('8', '<'))
```

Comment:
This would break the test on Fedora, since it check the version without restricting it to RHEL/Centos/etc. IE, Fedora 22 wouldn't run that test. I think we should have 2 includes, rather than merge them.